src/analytics/navigationTracker/navigationTracker.ts

```diff
@@ -34,15 +34,19 @@ export class NavigationTracker extends BaseTracker {
     if (element.tagName !== "A" && element.tagName !== "BUTTON") {
       return false;
     }
+    // Ignore links that don't have an href
+    if (!element.href || element.href === "") {
+      return false;
+    }
```

Comment:
Is there a requirement for this? If not then worth checking with the analysts if they want to see anything for incorrectly structure links

Reply:
It's more technical error handling that a product requirement.
I just want to be sure we are not using the navigation tracker if the link hasn't any "destination".
"a" tag can be use sometime just for triggering an onclick event (so a js function).
I just improved the conditions. element href can be empty, or doesn't exist , or be equal to "#" (no destination)

Reply:
Double checking as I'm about to add in a small change ticket to the backlog off the back of a conversation with Sam. They need all missing data to be replaced with `"undefined"` specifically but you raise a good point about the JS function use case

Reply:
When I built the tracker, I checked the links in a lot of pages from HRMC pages ;-)